### PR TITLE
Sanitize request body during capturing instead of serialization #656

### PR DESCRIPTION
**Pull request by gregkalapos**
We have some issues with our request body sanitization logic - currently the code has bugs and in some cases the whole body is incorrectly "sanitized away".

Prior to this PR we sanitized the request body during serialization. The reason for doing so was that if we sanitize during serialization then it applies to everything - including body captured manually through public API, ASP.NET Core and old ASP.NET - so 1 code for all. The challenge is that with this we only had a string and we tried to parse the keys and values from a single string - very resource consuming and hard to do it correctly. I tried to fix it in #617, but that still is not 100% correct.

In the meantime errors were reported multiple times: #656, #710. 

So, after some iteration, I decided that the best is to just sanitize the body in ASP.NET Core during capturing - this is trivial, there is a nice API for doing this, in that case we have the keys and values from the form encoded body - and then once the body is set we leave it as it is, so nothing happens during serialization.

Of course it'd be nice to have 1 implementation for body sanitization that is shared across all technologies, but I feel it's not worth the very high effort to make that implementation bug free.

### CODE REVIEW

**Comment by gregkalapos** [comment]
@bmorelli25 here we go again 😆; there is a new note in `docs/configuration.asciidoc`.

**Comment by gregkalapos** [comment]
Alternatively we could store the content type and do the sanitization based on that during serialization.

Reason we I did not do that:
- with public API it's not guaranteed that it's set 
- still during serialization with that we have to re-parse the body, which is resource consuming

Doing it during capturing is significantly simpler.

File: docs/configuration.asciidoc [hunk 1]
```diff
@@ -298,6 +298,8 @@ Sometimes it is necessary to sanitize, i.e., remove, sensitive data sent to Elas
 This config accepts a list of wildcard patterns of field names which should be sanitized.
 These apply for example to HTTP headers and `application/x-www-form-urlencoded` data.
 
+NOTE: In case you capture the request body manually through the public API, the agent won't sanitize the body - this setting only applies to values that are captured automatically by the agent.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc]: Let's bump this from a NOTE to IMPORTANT. Also, I think it makes sense to swap the sentences. What do you think of this?
```suggestion
IMPORTANT: This setting only applies to values that are captured automatically by the agent. If you capture the request body manually with the public API, this configuration doesn't apply, and the agent won't sanitize the body. 
```

Comment (gregkalapos) [reply docs/configuration.asciidoc]: Agreed 👍

**Comment by gregkalapos** [comment]
Failing test is due to unavailable `http://httpstat.us` - merging now.

**Pull request merged**